src/chainstate/stacks/boot/pox.clar

```diff
@@ -0,0 +1,754 @@
+        ;; depends on the burnchain being used.  When Bitcoin is
+        ;; the burnchain, this gets translated into a p2pkh, p2sh,
+        ;; p2wpkh-p2sh, or p2wsh-p2sh UTXO, depending on the version.
+        (pox-addr (tuple (version uint) (hashbytes (buff 20))))
+        ;; how long the uSTX are locked, in reward cycles.
+        (lock-period uint)
+        ;; reward cycle when rewards begin
+        (first-reward-cycle uint)
+        ;; Who, if anyone, is the delegate for this account?
+        ;; The delegate may begin Stacking on behalf of the
+        ;; locked-up STX tokens.
+        (delegate (optional principal))
+    )
+)
+
+;; How many uSTX are stacked in a given reward cycle.
+;; Updated when a new PoX address is registered, or when more STX are granted
+;; to it.
+(define-map reward-cycle-total-stacked
+    ((reward-cycle uint))
+    ((total-ustx uint))
+)
+
+;; Internal map read by the Stacks node to iterate through the list of
+;; PoX reward addresses on a per-reward-cycle basis.
+(define-map reward-cycle-pox-address-list
+    ((reward-cycle uint) (index uint))
+    (
+        (pox-addr (tuple (version uint) (hashbytes (buff 20))))
+        (total-ustx uint)
+    )
+)
+
+(define-map reward-cycle-pox-address-list-len
+    ((reward-cycle uint))
+    ((len uint))
+)
+
+;; When is a PoX address active?
+;; Used to check of a PoX address is registered or not in a given
+;; reward cycle.
+(define-map pox-addr-reward-cycles
+    ((pox-addr (tuple (version uint) (hashbytes (buff 20)))))
+    (
+        (first-reward-cycle uint)
+        (num-cycles uint)
+    )
+)
+
+;; Binding between the principal whose tokens are stacked, and the principal who
+;; is allowed to set the PoX reward address.  This binding also includes permissions
+;; that the delegator has, as described below.
+(define-map delegates
+    ;; The address of who owns the tokens to be stacked.
+    ((stacker principal))
+    (
+        ;; The address of the delegate, who stacks them.
+        (delegate principal)
+        ;; Number of uSTX this delegate may stack on this stacker's behalf.
+        (amount-ustx uint)
+    )
+)
+
+;; Binding between the delegate, and the amount of STX and PoX addresses it 
+;; is entrusted with.  Updated when the client delegates STX.
+(define-map delegate-control
+    ((delegate principal))
+    (
+        ;; Total number of uSTX locked to this delegate
+        (total-ustx uint)
+        ;; PoX address the delegate must use.
+        (pox-addr (tuple (version uint) (hashbytes (buff 20))))
+        ;; Earliest time at which the delegate can Stack the delegated STX.
+        (burn-block-height-start uint)
+        ;; Beginning of this delegate's tenure
+        (first-reward-cycle uint)
+        ;; How long the uSTX will be locked for, in reward cycles
+        (tenure-cycles uint)
+    )
+)
+
+;; Amount of uSTX that reject PoX, by reward cycle
+(define-map stacking-rejection
```

Comment:
The `(block-at ...)` feature was introduced for simplifying this kind of use cases. You could have a `data-var`, and reading the value at the height of the anchor-block.

Reply:
The reason it's written this way is because the queries on Stacking rejections concern whether or not PoX is happening in a particular reward cycle, not a particular block.